### ISSUE TITLE
Where's the encoder_freeze parameter?

### ISSUE THREAD

**Issue by ricber** [qubvel-org/segmentation_models.pytorch#1197]
Hello,

I've read in some previous issues that it's possible to specify the parameter encoder_freeze to freeze the encoder weights. However, I couldn't find this parameter anywhere in the repository, and it's also not mentioned in the documentation.
Is this parameter still valid? If so, why doesn't it appear in the code?

Thank you!

**Comment by qubvel** [comment]
Hello, this is not smth that was added to the API, but you can always do it manually:

```py
for param in model.encoder.parameters():
    param.requires_grad = False
```

P.S. It might be a bit more complicated for batchnorm layers. Would appreciate a PR in case you have bandwidth! 🤗

**Comment by ricber** [comment]
What’s your idea? To introduce an `encoder_freeze` parameter in the API? If so, in which class should the parameter be introduced?

**Comment by qubvel** [comment]
This method could be added to a base `SegmentationModel` class
